zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -239,7 +241,8 @@ private[consumer] final class Runloop private (
         } yield ()
     )
 
-    recordRebalanceRebalancingListener ++ settings.rebalanceListener
+    (recordRebalanceRebalancingListener ++ settings.rebalanceListener.runOnExecutor(topLevelExecutor))
+      .toKafka(sameThreadRuntime, rc)
```

Comment:
What do you think of something like this:

```suggestion
    // Here we just want to avoid any executor shift if the user provided listener is the noop listener.
    val userRebalanceListener =
      settings.rebalanceListener match {
        case RebalanceListener.noop => RebalanceListener.noop
        case _                      => settings.rebalanceListener.runOnExecutor(topLevelExecutor)
      }

    (recordRebalanceRebalancingListener ++ userRebalanceListener).toKafka(sameThreadRuntime, rc)
```

Reply:
Most people will not install a user rebalance listener so I like this idea.

Reply:
This is implemented now.